pkg/serverless/trace/inferredspan/span_enrichment.go

```diff
@@ -217,10 +220,120 @@ func (inferredSpan *InferredSpan) EnrichInferredSpanWithS3Event(eventPayload eve
 	}
 }
 
+type DatadogPayload struct {
+	Type        string `json:"Type"`
+	Value       string `json:"Value"`
+	DataType    string `json:"dataType"`
+	BinaryValue string `json:"binaryValue"`
+	StringValue string `json:"stringValue"`
+}
+
+type TraceHeader struct {
+	TraceID          string `json:"x-datadog-trace-id"`
+	ParentID         string `json:"x-datadog-parent-id"`
+	SamplingPriority string `json:"x-datadog-sampling-priority"`
+	Sampled          string `json:"x-datadog-sampled"`
+}
+
+type customMessageAttributeStruct struct {
+	Type  string `json:"Type"`
+	Value string `json:"Value"`
+}
+type BodyStruct struct {
+	Message           string                                  `json:"Message"`
+	MessageAttributes map[string]customMessageAttributeStruct `json:"MessageAttributes"`
+}
+type CustomDatadogPayload struct {
+	Type        string `json:"Type"`
+	Value       string `json:"Value"`
+	DataType    string `json:"dataType"`
+	BinaryValue string `json:"binaryValue"`
+	StringValue string `json:"stringValue"`
+}
+
+func extractContextFromSNSSQSEvent(firstRecord events.SQSMessage) (string, string, string, error) {
```

Comment:
It's a code smell to return an untyped set of strings from a method like this.

If I didn't study this method's implementation, I'd have no idea that the first string is the TraceID, then the Parent ID, etc.

We should use a type here - we already have `TraceHeader`, maybe we need a `traceContext` struct? (again - check out the codebase and see if we have it already. I didn't see one though)